src/DGtal/geometry/surfaces/DigitalSurfaceConvolver.ih

```diff
@@ -1096,15 +1096,18 @@ DGtal::DigitalSurfaceConvolver< Functor, KernelFunctor, KSpace, DigitalKernel, 2
               {
                 for( KernelConstIterator itm = myKernelMask->first; itm != myKernelMask->second; ++itm )
                   {
-                    myKSpace.sSetKCoords( shiftedSpel, myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel );
+                    if( myKSpace.cIsInside( myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel ))
+                    {
+                      myKSpace.sSetKCoords( shiftedSpel, myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel );
```

Comment:
Agree, I need to check this a little more to be compatible with periodic dimension. I can't do this today, but tomorrow I think.